File: quantile/templates/test/fixtures/_test.matrix.R
options( digits = 16 )
library( jsonlite )

<%= parameters.map( function( p ) { return p.name + ' = ' + p.default } ).join( '\n' ) %>
probs = 0:24 / 25
y = <%= rName %>( probs, <%= parameters.map( function( p ) { return p.name } ).join( ', ' ) %> )

cat( y, sep = ",\n" )

data = list(
	<%= parameters.map( function( p ) { return p.name + ' = ' + p.name } ).join( ',\n\t' ) %>,
	data = probs,
	expected = y
)


write( toJSON( data, digits = 16, auto_unbox = TRUE ), "./test/fixtures/matrix.json" )

File: mgf/templates/test/fixtures/_test.matrix.jl
using Distributions
using JSON

<%= parameters.map( function( p ) { return p.name + ' = ' + p.default } ).join( '\n' ) %>
d = <%= rName %>( <%= parameters.map( function( p ) { return p.name } ).join( ',' ) %> )

x = linspace( 1, 0, 25 )

dmgf(t) = mgf(d, t )
y = map( dmgf, x )
println( y )

data = Dict([
	<%- parameters.map( function( p ) { return '("' + p.name + '", ' + p.name + ")" } ).join( ',\n\t' ) %>,
	("data", x),
	("expected", y)
])

outfile = open("./test/fixtures/matrix.json", "w")
JSON.json(data)

write( outfile, JSON.json(data) )
